perceval/backends/remo2.py

```diff
@@ -0,0 +1,318 @@
+            items_data = json.loads(raw_items)
+            titems = items_data['count']
+
+            items = items_data['results']
+            for item in items:
+                raw_item_details = self.client.call(item['_url'])
+                self._push_cache_queue(raw_item_details)
+                item_details = json.loads(raw_item_details)
+                yield item_details
+                nitems += 1
+
+                self._flush_cache_queue()
+
+        logger.info("Total number of events: %i (%i expected)", nitems, titems)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the items from the cache.
+
+        :returns: a generator of items
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+
+        logger.info("Retrieving cached ReMo items: '%s'", self.url)
+
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        nitems = 0
+
+        while True:
+            data = json.loads(next(cache_items))
+            # The raw_data is always a list of items or an item
+            if 'count' in data:
+                # It is a list
+                continue
+            else:
+                yield data
+                nitems += 1
+
+        logger.info("Retrieval process completed: %s items retrieved from cache",
+                    nitems)
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an event item."""
+        return str(item['remo_url'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a ReMo item.
+
+        The timestamp is extracted from 'end' field.
+        This date is converted to a perceval format using a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+
+        if 'end' in item:
+            # events updated field
+            updated = item['end']
+        elif 'date_joined_program' in item:
+            # users updated field that always appear
+            updated = item['date_joined_program']
+        elif 'report_date' in item:
+            # activities updated field
+            updated = item['report_date']
+        else:
+            raise RuntimeError("Can't find updated field for item %s", item)
+
+        return float(str_to_datetime(updated).timestamp())
+
+
+class ReMoClient:
+    """ReMo API client.
+
+    This class implements a simple client to retrieve events from
+    projects in a ReMo site.
+
+    :param url: URL of ReMo (sample https://reps.mozilla.org)
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, url, limit=40):
+        # 40 it the max limit for events.
+        self.url = url
+        self.limit = limit
+        self.api_activities_url = urljoin(self.url, '/api/beta/activities/')
```

Comment:
Refactored as:

`self.api_activities_url = urljoin(self.url, ReMoClient.API_PATH+'/activities/')`

ReMoClient.API_PATH is probably the string that is going to change. Are you comfortable with that?